Fix Grammar Ambiguity

vihanb opened the pull request:
Attempts to fix points of ambiguity within nearley grammar.

vihanb committed:
Added codeclimate

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
test: added eslint & codeclimate w/ it.

This adds ESLint and CodeClimate integration to better check the
code quality

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Explicitly specify .eslintrc.yml for codeclimate

Looks like CodeClimate wasn't picking it up so just adding this to test

vihanb committed:
Progress commit which trys to fix some of borks

Appears that some of the borks are still there, particullarially
in binary expressions.

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Moved eslintrc to JS.

I was originall using yml for consistency but looks like eslint
likes JS more so why not.

Signed-off-by: Vihan B <contact@vihan.org>

taylorhansen commented on src/vsl/parser/grammar/expr.ne:
```diff
@@ -27,33 +24,43 @@ function recursiveProperty(head, tail, optional, location) {
+                  | ArgumentCallHead ("," _ ArgumentCall {% nth(2) %}):+ {% (d, l) => new t.FunctionCall(null, [d[0]].concat(d[1]), false, l) %}
+                  | ArgumentCallHead {%
+                  // O: KEEP
+                    (d, l) => console.log('asadsdasfajf', t.Tuple) || d[0] instanceof t.Identifier ? new t.PropertyExpression(null, d[0], false, l) : d[0] instanceof t.Tuple ? new t.FunctionCall(null, console.log(d[0].tuple) || d[0].tuple.map(item => new t.ArgumentCall(item, null, item.position)), l) : new t.FunctionCall(null, [d[0]], false, l)
```
These lines are waaay too long D: it seems like a lot of the grammar borks come from this issue because they're hard to read that way.

vihanb replied:
We can't do much about the postprocessing line length but pls see @somebody1234 for all nearley related question/comment

vihanb replied:
@CrazyGuy108 I've made new argument stuffs which is hopefully easier to read :D

vihanb committed:
Fixed ambiguity by redoing function calls.

I took out command chains but they have design issues that need to
be addressed before we start implementing them.

This fixes ambiguity which arised within function call expressions
along with ones within addition. Command chains viewed `1 + 1` as:

    1(+1)

which caused this issue. Additionally the `ArgumentCallHead` had
unusual construction so I've removed that (or at least disabled) so
we have a working parser now. Hopefully this will allow all the
tests and all to pass and we can continue dev.

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Merge branch 'develop' into fix/grammar-ambiguity

vihanb committed:
Merge branch 'vihanb/codeclimate' into fix/grammar-ambiguity

vihanb committed:
Fixed bug with codecov doing a weird.

Signed-off-by: Vihan B <contact@vihan.org>

vihanb committed:
Merge branch 'fix/grammar-ambiguity' of github.com:vsl-lang/VSL into fix/grammar-ambiguity

vihanb committed:
I am idiot ok now to wait 8 more minutes

Signed-off-by: Vihan B <contact@vihan.org>
